map/src/main/kotlin/edu/artic/map/GoogleMapUtil.kt

```diff
@@ -79,8 +76,32 @@ internal class TransparencyProperty : Property<TileOverlay, Float>(Float::class.
  * @param bounds the restrictions of
  * [the map's viewport][com.google.android.gms.maps.Projection.getVisibleRegion]
  */
-fun LatLng.isCloseEnoughToCenter(bounds: LatLngBounds): Boolean {
-    return bounds.contains(this) && bounds.center.distanceTo(this) < 15
+fun LatLng.isCloseEnoughToCenter(region: VisibleRegion): Boolean {
+    val bounds = region.latLngBounds
+    val nearLeft = region.nearLeft
+    val nearRight = region.nearRight
+    val farLeft = region.farLeft
+    val farRight = region.farRight
+    val nearDistance = nearLeft.distanceTo(nearRight)
+    val farDistance = farLeft.distanceTo(farRight)
+
+    val latDiffFar = (farRight.latitude - farLeft.latitude)/2f
+    val lonDiffFar = (farRight.longitude - farLeft.longitude)/2f
+    val centerFar = LatLng(farLeft.latitude + latDiffFar, farLeft.longitude + lonDiffFar)
```

Comment:
Could we have a function `midpoint(first: LatLng, second: LatLng)` that did the calculation for `centerFar` and `centerNear`?